### ISSUE TITLE
Start/Stop/Pause Algorithms

### ISSUE THREAD

**Issue by SigmaX** [AureumChaos/LEAP#154]
Right now, if clients wanted to be able to pause and continue an algorithm (for example, as part of a GUI application), they could write their own code to wrap, say 1) a custom evolutionary loop that checks for pause/stop messages at each iteration, or 2) one of our `leap_ec.algorithm` metaheuristics, which returns an iterator that can be used in a similar way.

It would be nice if we offered a ready-made wrapper to do this, however: something people could easily use to control LEAP via GUI components or what-naught.

Bonus if it works nicely with parallel threads.

**Comment by piprrr** [comment]
I think you're referring to checkpoints and restarts?  I *thought* we had an issue for that, but I was surprised to see we hadn't.  

Checkpoints and restarts would be a critical feature, especially for HPC-related work.  E.g., on Summit it will be important to checkpoint and restart when jobs exhaust their budgeted time and restart later.

**Comment by SigmaX** [comment]
Slightly different.  Checkpoints and restarts fit the HPC need, to keep from losing work during long-running jobs.

Being able to send start/stop/pause could be something you wire up to GUI buttons labelled "Start," "Stop," and "Pause"—so 1) responding to user-generated events, and 2) without necessarily saving anything to a file (ex. just pause in memory and wait until the user hits play!).

**Comment by SigmaX** [comment]
(This issue is motivated by a project colleagues are working on right now, where they've been frustrated that other EA frameworks lack this feature).

**Comment by piprrr** [comment]
Fair enough.  In which case I will add a separate, but possibly related, issue for adding checkpoint/restart support.
